src/components/PageRefreshButton/PageRefreshButton.tsx

```diff
@@ -1,20 +1,25 @@
 import { useQueryClient } from '@tanstack/react-query';
-import { useState } from 'react';
+import { forwardRef, useEffect, useRef, useState } from 'react';
 import { toast } from 'sonner';
 import logger from '@/logger';
 import { RefreshCw } from 'lucide-react';
 import { cn } from '@/lib/utils.ts';
 import { Button } from '@/components';
 
 export const PageRefreshButton = () => {
+  const refreshButtonRef = useRef<HTMLButtonElement>(null);
+
+  return <RefreshButtonComponent ref={refreshButtonRef} />;
+};
+
+export const RefreshButtonComponent = forwardRef<HTMLButtonElement>((_, ref) => {
   const queryClient = useQueryClient();
   const [isRefreshing, setIsRefreshing] = useState(false);
+  const [catchEye, setCatchEye] = useState(false);
 
   const onRefreshClick = async () => {
     setIsRefreshing(true);
     try {
-      // Wait for all queries invalidation (data refetching)
-      // and a minimum delay of 0.5sec before proceeding to make sure the UI doesn't flicker
```

Comment:
Why you've removed my comment? 
I found it explanatory 🙈

Reply:
it's not me, it's prettier 🙈

Reply:
fixed